src/types.ts

```diff
@@ -221,13 +217,52 @@ export type ChatCompletionOptionsFunction = {
   parameters: ObjectSchema;
 };
 
-type JSONSchema = (
-  | ObjectSchema
-  | StringSchema
-  | NumberSchema
-  | BooleanSchema
-  | ArraySchema
-) & { description?: string };
+interface SystemCompletionMessage {
+  content: string;
+  role: "system";
+}
+
+interface UserCompletionMessage {
+  content: string;
+  role: "user";
+}
+
+interface AssistantCompletionMessage {
+  content: string;
+  role: "assistant";
+}
+
+interface FunctionAwareAssistantCompletionMessage {
+  content: string | null;
+  role: "assistant";
+  function_call?: {
+    "name": string;
+    "arguments": string;
+  };
+}
+
+interface FunctionCompletionMessage {
+  content: string;
+  role: "function";
+  name: string;
```

Comment:
Are we sure name can only show up here? If yes, LGTM! Thanks!

Reply:
Good question.  The [main docs](https://platform.openai.com/docs/api-reference/chat/create#chat/create-name) really only refers to name as something used by function, BUT while trying to verify this I stumbled on examples from 
[the cookbook](https://github.com/openai/openai-cookbook/blob/main/examples/How_to_format_inputs_to_ChatGPT_models.ipynb) using message name to further instruct system (Cases 10 and 13).

With that said, in [f81bdd8](https://github.com/load1n9/openai/pull/19/commits/f81bdd8baca59bf5743cfcc29dd53804a1234ca3), I marked them as optional again.

Reply:
Thanks, let me verify that this is still fine and I'll merge.